### DISCUSSION TITLE
assemble data from non-continuous bits?

### DISCUSSION THREAD

**Discussion by knarrff** [merbanan/rtl_433#2182] [Q&A]
Hi,

I two door bells that use PCM and transmit bits like in these examples (two examples, one by each bell):
```
1001001001001011001001001011001011001
1001011011001011001011011011011001001
```
Within these, only some bits are actual information, the rest is ... just there for some reason. Let's write those two lines differently:
```
(00)1 001 001 001 001 011 001 001 001 011 001 011 001
(00)1 001 011 011 001 011 001 011 011 011 011 001 001
       ^   ^   ^   ^   ^   ^   ^   ^   ^   ^   ^   ^
```
As you can see: the first bit in each group is always
 a 0, the third is always a 1. The remaining 12 bits (not counting the initial (00)1, as this is likely just a preamble) gives 12 bits, enough to transmit an 8-bit ID and a 3/4-bit bell tone. Those bits are what I am interested in (marked with ```^``` above). The first 8 of those turn out to be the ID, the remaining 4 (or 3, not sure here) are the bell tone.

How can I extract just those bits and turn them into meaningful numbers (decimal/hex/bell tone name)? Can I do that using the flex decoder? Looking at the examples, it should be able to handle PCM just fine, but I don't know how to extract those "distributed" bits into one number. All the examples in /conf seem to use continuous bits and I could not really find a documentation of the "getter"-syntax either.

Thanks.

**Comment by knarrff** [comment]
Just in case this is useful: this is what the signal of one of these looks like:
https://triq.org/pdv/#AAB014030101300268125480909180918090909091818255+AAB015035B0130026812548180909180918090909091818255+AAB00D0301013002681254818090918055 . Maybe I am wrong about PCM and I should choose something else?

**Comment by zuckschwerdt** [comment]
Yes ;) that's PWM: https://triq.org/rtl_433/PULSE_FORMATS.html#fixed-bit-width-pwm-with-leading-gaps
